tck-test/SingleSubscriber.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "tck-test/SingleSubscriber.h"
+
+#include <thread>
+
+#include <folly/Format.h>
+
+using namespace folly;
+
+namespace rsocket {
+namespace tck {
+
+void SingleSubscriber::request(int n) {
+  LOG(INFO) << "... requesting " << n << ". No request() for Single.";
+}
+
+void SingleSubscriber::cancel() {
+  LOG(INFO) << "... canceling ";
+  canceled_ = true;
+  if (auto subscription = std::move(subscription_)) {
+    subscription->cancel();
+  }
+}
+
+void SingleSubscriber::onSubscribe(
+    yarpl::Reference<yarpl::single::SingleSubscription> subscription) noexcept {
+  VLOG(4) << "OnSubscribe in SingleSubscriber";
+  subscription_ = subscription;
+}
+
+void SingleSubscriber::onSuccess(Payload element) noexcept {
+  LOG(INFO) << "... received onSuccess from Publisher: " << element;
+  {
+    std::unique_lock<std::mutex> lock(mutex_);
+    std::string data =
+        element.data ? element.data->moveToFbString().toStdString() : "";
+    std::string metadata = element.metadata
+        ? element.metadata->moveToFbString().toStdString()
+        : "";
+    values_.push_back(std::make_pair(data, metadata));
+    ++valuesCount_;
+  }
+  valuesCV_.notify_one();
+  {
+    std::unique_lock<std::mutex> lock(mutex_);
+    completed_ = true;
```

Comment:
completed_ is already std::atomic, do we need to lock a mutex to update its value?